src/properties/propertiesService.ts

```diff
@@ -27,6 +29,38 @@ interface PropertySchemas {
   uiSchema?: UISchemaElement;
 }
 
+const ruleSchema = {
+  rule: {
+    type: 'object',
+  },
+};
+
+const ruleUiSchema = {
+  type: 'Control',
+  scope: '#/properties/rule',
+};
+
+const withRule = (schemas: PropertySchemas) => {
+  const schema = schemas.schema.properties
+    ? schemas.schema
+    : {
+        type: 'object',
+        properties: {},
+      };
+  const uiSchema =
+    schemas.uiSchema && isLayout(schemas.uiSchema)
+      ? schemas.uiSchema
+      : {
+          type: 'VerticalLayout',
+          elements: [],
+        };
+
+  assign(schema.properties, ruleSchema);
+  (uiSchema as Layout).elements.push(ruleUiSchema);
+
+  return { schema, uiSchema };
+};
+
 export class ExamplePropertiesService implements PropertiesService {
```

Comment:
Somehow I missed this before, but this should not be called "ExamplePropertiesService" but for example `PropertiesServiceImpl`